geo/src/algorithm/contains/polygon.rs

```diff
@@ -0,0 +1,91 @@
+use super::Contains;
+use crate::intersects::Intersects;
+use crate::kernels::*;
+use crate::*;
+
+// ┌─────────────────────────────┐
+// │ Implementations for Polygon │
+// └─────────────────────────────┘
+
+impl<T> Contains<Coordinate<T>> for Polygon<T>
+where
+    T: HasKernel,
+{
+    fn contains(&self, coord: &Coordinate<T>) -> bool {
+        match utils::coord_pos_relative_to_ring(*coord, &self.exterior()) {
+            utils::CoordPos::OnBoundary | utils::CoordPos::Outside => false,
+            _ => self.interiors().iter().all(|ls| {
+                utils::coord_pos_relative_to_ring(*coord, ls) == utils::CoordPos::Outside
+            }),
+        }
+    }
+}
+
+impl<T> Contains<Point<T>> for Polygon<T>
+where
+    T: HasKernel,
+{
+    fn contains(&self, p: &Point<T>) -> bool {
+        self.contains(&p.0)
+    }
+}
+
+// TODO: ensure DE-9IM compliance: esp., when
+// line.start and line.end is on the boundaries
+impl<T> Contains<Line<T>> for Polygon<T>
+where
+    T: HasKernel,
+{
+    fn contains(&self, line: &Line<T>) -> bool {
+        // both endpoints are contained in the polygon and the line
+        // does NOT intersect the exterior or any of the interior boundaries
+        self.contains(&line.start)
+            && self.contains(&line.end)
+            && !self.exterior().intersects(line)
+            && !self.interiors().iter().any(|inner| inner.intersects(line))
+    }
+}
+
+// TODO: also check interiors
```

Comment:
Right, this one also looks wrong. I wouldn't want to include new broken implementations, but I believe you are just moving these, right?

Reply:
Yeah, these implementations were just moved into this file.  We should fix the two traits for de9im compliance in another pr